### PR TITLE
AADGroup: Implements #2301

### PR DESCRIPTION
**Pull request by salbeck-sit**
#### Pull Request (PR) description
AADGroup: Add properties MemberOf and AssignedToRole to resource

#### This Pull Request (PR) fixes the following issues
Implements #2301

### CODE REVIEW

**Comment by NikCharlebois** [comment]
@salbeck-sit can you please take a look at the failing Unit Tests? Thanks

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Small typo. Can you also add an entry in the ChangeLog? Thanks

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1 [hunk 1]
```diff
@@ -164,7 +172,7 @@ function Get-TargetResource
             if ($Group.MembershipRuleProcessingState -ne 'On')
             {
                 # Members
-                [Array]$members = Get-MgGroupMember -GroupId $Group.Id -All:$true
+                [Array]$members = Get-MgGroupMember -GroupId $Group.Id
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1:175]: Why are we removing the -All here? It should be included to make sure we catch all members

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1 (hunk 1 above)
Comment (salbeck-sit) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1:175]: -ALL is apparently no longer supported by Get-MgGroup* cmdlets in v1.12.0 of the Microsoft.Graph.Groups module. 
Running the unit-tests with -All parameters resulted in a number of 'caught throws' where the text in the corresponding M365DSC eventlog stated that "A parameter cannot be found that matches parameter name 'All'". The only cmdlets in the Get-TargetResource function that used -All were the Get-MgGroup* cmdlets. Removing -All everywhere resulted in a passed unit-test.

The question of whether this should have been brought to the attention of the folks responsible for Graph-modules is another matter. I guess you could have gotten them to update the relevant module in a day or two as something similar has happened before, but I really just wanted to get the change under wraps.

Comment (andikrueger) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1:175]: There was an intermediate release. Graph modules at up to 1.12.1 now and do allow the -all parameter

Comment (andikrueger) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1:175]: See https://github.com/microsoft/Microsoft365DSC/commit/e01d15b4d208a1fa666afd01f25c423323b87816

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**